scripts/components/pixel-canvas.js

```diff
@@ -35,6 +35,56 @@ class PixelCanvasProto extends HTMLElement {
   }
 
   /**
+    Fetch the current canvas state and turn it into a
+    JSON string representing the canvas.
+
+    First we turn the data into a 2D array where each
+    entry is the color hash of the pixels current
+    background color. Then we JSON.stringify that array.
+
+    @return {string} - a string produced by calling JSON.strigify on
+      a 2D Array containing color hex values, such as '#ffffff' for white.
+  */
+  serializeGrid() {
+    let grid = [];
+
+    for(let h = 0; h < this.height; h++) {
+      let row = [];
+      grid[h] = row;
+
+      for(let w = 0; w < this.width; w++) {
+
+          let offset = 2; // Never look where the grid-lines are drawn
+          let x = (this.pixelSize * w) + offset;
+          let y = (this.pixelSize * h) + offset;
+
+          // TODO: may want to pull this into a function
+          let [red, green, blue] = this.canvasCtx.getImageData(x, y, 1, 1).data;
```

Comment:
^^ @jamesont has it -- essentially the canvasCtxt is created by the canvas API elsewhere in the code, here are the most important two lines:

```
    this.canvas = document.createElement('canvas');
    this.canvasCtx = this.canvas.getContext('2d');
```

Create a new HTML Canvas Element, then use the "getContext" method to get the `context`. The `context` object is used for drawing in the canvas (among other things)